File: check_remaining.js
const { Client } = require('pg');
const client = new Client({ connectionString: 'postgresql://postgres:bJmkMzZQLqhzYltMNDwentYAPFUAppSq@tramway.proxy.rlwy.net:39193/railway' });

async function run() {
  await client.connect();
  
  // NounsForms: ID, RawyID, RawyText, RawyTextID, RawyTextShape, Frequency
  // Check narrators table for RawyTextShape, RawyTextID-like columns
  console.log('=== narrators sample row ===');
  const ns = await client.query(`SELECT * FROM narrators LIMIT 1;`);
  console.log(JSON.stringify(ns.rows[0]));
  
  // NounsStat1: ID, Title, Count
  // Check if narrator_books table exists or has stat data
  console.log('\n=== narrator_books columns ===');
  const nb = await client.query(`SELECT column_name, data_type FROM information_schema.columns WHERE table_schema='public' AND table_name='narrator_books' ORDER BY ordinal_position;`);
  nb.rows.forEach(c => console.log(`  ${c.column_name}: ${c.data_type}`));
  const nbCount = await client.query(`SELECT COUNT(*) FROM narrator_books;`);
  console.log(`narrator_books count: ${nbCount.rows[0].count}`);
  const nbSample = await client.query(`SELECT * FROM narrator_books LIMIT 3;`);
  console.log('SAMPLE:', JSON.stringify(nbSample.rows));
  
  // HadithJudgmentScientists: ID, Name
  // hadith_judgments has scientist_id - is there a separate scientist table?
  // narrator_scientists has scientist_id, scientist_name
  console.log('\n=== narrator_scientists SAMPLE ===');
  const nsc = await client.query(`SELECT * FROM narrator_scientists LIMIT 3;`);
  console.log(JSON.stringify(nsc.rows));
  
  // ExpRawyModbaj: PrID, RawyID, RName, ShName, ShyoukhID, Dic
  // This looks like a denormalized narrator-sheikh join
  // Check isnad_relations or narrator_relations for this
  console.log('\n=== isnad_relations SAMPLE (first 3) ===');
  const ir = await client.query(`SELECT column_name, data_type FROM information_schema.columns WHERE table_schema='public' AND table_name='isnad_relations' ORDER BY ordinal_position;`);
  ir.rows.forEach(c => console.log(`  ${c.column_name}: ${c.data_type}`));
  const irCount = await client.query(`SELECT COUNT(*) FROM isnad_relations;`);
  console.log(`isnad_relations count: ${irCount.rows[0].count}`);
  
  await client.end();
}
run().catch(e => { console.error(e.message); process.exit(1); });

File: check_remaining.py
import psycopg2, sys, io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')
DB = "postgresql://postgres:bJmkMzZQLqhzYltMNDwentYAPFUAppSq@tramway.proxy.rlwy.net:39193/railway"
conn = psycopg2.connect(DB, sslmode="require")
cur = conn.cursor()

# Check tables that don't have pages yet
tables_to_check = [
    'hadith_controversial_tree', 'hadith_controversial_descriptions',
    'lexicon_categories', 'lexicon_items', 'lexicon_hadith',
    'hadith_shawahed',
    'amthal', 'gwamh', 'gwamh_items',
    'matn_comparison',
    'hadith_index_categories', 'hadith_index_items',
]

for t in tables_to_check:
    cur.execute(f"SELECT COUNT(*) FROM {t}")
    n = cur.fetchone()[0]
    print(f"{t}: {n}")

# Check lexicon structure
cur.execute("SELECT column_name FROM information_schema.columns WHERE table_name='lexicon_categories' ORDER BY ordinal_position")
print("lexicon_categories cols:", [r[0] for r in cur.fetchall()])
cur.execute("SELECT * FROM lexicon_categories LIMIT 3")
for r in cur.fetchall():
    print("lex_cat:", r)

cur.execute("SELECT column_name FROM information_schema.columns WHERE table_name='lexicon_items' ORDER BY ordinal_position")
print("lexicon_items cols:", [r[0] for r in cur.fetchall()])
cur.execute("SELECT * FROM lexicon_items LIMIT 3")
for r in cur.fetchall():
    print("lex_item:", r)

cur.execute("SELECT column_name FROM information_schema.columns WHERE table_name='lexicon_hadith' ORDER BY ordinal_position")
print("lexicon_hadith cols:", [r[0] for r in cur.fetchall()])
cur.execute("SELECT * FROM lexicon_hadith LIMIT 3")
for r in cur.fetchall():
    print("lex_hadith:", r)

# Controversial tree
cur.execute("SELECT column_name FROM information_schema.columns WHERE table_name='hadith_controversial_tree' ORDER BY ordinal_position")
print("controversial_tree cols:", [r[0] for r in cur.fetchall()])
cur.execute("SELECT * FROM hadith_controversial_tree LIMIT 3")
for r in cur.fetchall():
    print("controversial:", r)

# amthal
cur.execute("SELECT column_name FROM information_schema.columns WHERE table_name='amthal' ORDER BY ordinal_position")
print("amthal cols:", [r[0] for r in cur.fetchall()])
cur.execute("SELECT * FROM amthal LIMIT 3")
for r in cur.fetchall():
    print("amthal:", r)

conn.close()
